src/modules/core/components/Fields/DatePicker/TimePicker.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { format } from 'date-fns/esm';
+import { defineMessages } from 'react-intl';
+import { flip, useFloating } from '@floating-ui/react-dom';
+import classnames from 'classnames';
+
+import Icon from '~core/Icon';
+import styles from './DatePicker.css';
+import { DEFAULT_TIME_FORMAT } from './constants';
+
+const MSG = defineMessages({
+  expandIconHTMLTitle: {
+    id: 'DatePicker.TimePicker.expandIconHTMLTitle',
+    defaultMessage: 'expand',
+  },
+});
+
+const isSameTimeOption = (dateA: Date, dateB: Date) =>
+  dateA.getUTCHours() === dateB.getUTCHours() &&
+  dateA.getUTCMinutes() === dateB.getUTCMinutes();
+
+interface Props {
+  selectedDate: Date;
+  onChange: (date: Date) => void;
+  timeInterval: number;
+}
+
+const TimePicker = ({ selectedDate, onChange, timeInterval }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { x, y, reference, floating, strategy, update } = useFloating({
+    placement: 'bottom',
+    middleware: [flip()],
+  });
+
+  /** Recalculate time dropdown positioning when selectedDate changes */
+  useEffect(() => {
+    update();
+  }, [selectedDate, update]);
+
+  /** Scroll the selected option into view upon opening the dropdown */
+  const selectedOptionRef = useRef<HTMLButtonElement | null>(null);
+  useEffect(() => {
+    if (!isOpen || !selectedOptionRef.current) {
+      return;
+    }
+
+    selectedOptionRef.current.scrollIntoView();
+  }, [isOpen]);
+
+  const formattedTime = useMemo(
+    () => format(selectedDate, DEFAULT_TIME_FORMAT),
+    [selectedDate],
+  );
+
+  const timeOptions = useMemo(() => {
+    return [...new Array((24 * 60) / 30)].map((_, idx) => {
```

Comment:
Small one. `24 * 2` makes more sense to me intuitively. "24 hours in a day, times 2 to get half hours".

Reply:
I've only just realised there's a bug in this line. The value `30` should be `timeInterval` instead